alicloud/resource_alicloud_common_bandwidth_package_test.go

```diff
@@ -0,0 +1,185 @@
+				skip = false
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping Common Bandwidth Package: %s (%s)", name, id)
+			continue
+		}
+		log.Printf("[INFO] Deleting Common Bandwidth Package: %s (%s)", name, id)
+		req := vpc.CreateDeleteCommonBandwidthPackageRequest()
+		req.BandwidthPackageId = id
+		_, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+			return vpcClient.DeleteCommonBandwidthPackage(req)
+		})
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete Common Bandwidth Package (%s (%s)): %s", name, id, err)
+		}
+	}
+	return nil
+}
+
+func TestAccAlicloudCommonBandwidthPackage_basic(t *testing.T) {
+	var commonBandwidthPackage vpc.DescribeCommonBandwidthPackagesResponse
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		// module name
+		IDRefreshName: "alicloud_common_bandwidth_package.foo",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckCommonBandwidthPackageDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCommonBandwidthPackageConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCommonBandwidthPackageExists("alicloud_common_bandwidth_package.foo", &commonBandwidthPackage),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo", "bandwidth", "200"),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo", "name", "tf_testAcc_common_bandwidth_package"),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo", "description", "tf_testAcc_common_bandwidth_package"),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo", "internet_charge_type", "PayByBandwidth"),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo2", "internet_charge_type", "PayBy95"),
+					resource.TestCheckResourceAttr(
+						"alicloud_common_bandwidth_package.foo2", "ratio", "20"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCommonBandwidthPackageExists(n string, commonBandwidthPackage *vpc.DescribeCommonBandwidthPackagesResponse) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Common Bandwidth Package ID is set")
+		}
+		client := testAccProvider.Meta().(*connectivity.AliyunClient)
+		commonBandwidthPackageService := CommonBandwidthPackageService{client}
+		instance, err := commonBandwidthPackageService.DescribeCommonBandwidthPackage(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if commonBandwidthPackage == nil || len((*commonBandwidthPackage).CommonBandwidthPackages.CommonBandwidthPackage) <= 0 {
+			return err
+		}
+		(*commonBandwidthPackage).CommonBandwidthPackages.CommonBandwidthPackage[0].BandwidthPackageId = instance.BandwidthPackageId
+		return nil
+	}
```

Comment:
I think line 140 - 144 is redundant and DescribeCommonBandwidthPackage has done it.